marine.c

```diff
@@ -966,6 +980,363 @@ WS_DLL_PUBLIC void marine_free(marine_result *ptr) {
     }
 }
 
+static marine_packet_field *
+marine_packet_field_init(marine_packet_field *node, char *name) {
+    node->name = name;
+    node->children = NULL;
+    node->value.type = MARINE_VT_NONE;
+    node->value.len = 0;
+    return node;
+}
+
+static void
+marine_packet_field_add_child(marine_packet_field *parent, marine_packet_field *child) {
+    if (parent->children == NULL) {
+        parent->children = g_array_new(TRUE, TRUE, sizeof(marine_packet_field));
+    }
+    g_array_append_val(parent->children, *child);
+}
+
+static marine_packet_field_value
+marine_packet_field_load_value(field_info *finfo) {
+    ftenum_t ftype = finfo->value.ftype->ftype;
+    marine_packet_field_value value;
+    if (IS_FT_INT(ftype)) {
+        value.type = MARINE_VT_INT;
+        value.len = finfo->value.ftype->wire_size;
+        if (IS_FT_INT32(ftype)) {
+            value.int_value = fvalue_get_sinteger(&finfo->value);
+        }
+        if (IS_FT_INT64(ftype)) {
+            value.int_value = fvalue_get_sinteger64(&finfo->value);
+        }
+    } else if (IS_FT_UINT(ftype)) {
+        value.type = MARINE_VT_UINT;
+
+        value.len = finfo->value.ftype->wire_size;
+        if (IS_FT_UINT32(ftype)) {
+            value.uint_value = fvalue_get_uinteger(&finfo->value);
+        }
+        if (IS_FT_UINT64(ftype)) {
```

Comment:
I'd use `else if ` here, and maybe even add an `else` case to have an explicit API (in case of error maybe? set to NONE and len 0?)

Same comment applies to the signed case